app/javascript/hyacinth_ui_v1/components/digital_objects/new/DigitalObjectNewForm.js

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import { useParams } from 'react-router-dom';
+import { useQuery } from '@apollo/react-hooks';
+import { capitalize } from 'lodash';
+
+import ContextualNavbar from '../../layout/ContextualNavbar';
+import { Can } from '../../../util/ability_context';
+import { getProjectQuery } from '../../../graphql/projects';
+import GraphQLErrors from '../../ui/GraphQLErrors';
+import MetadataForm from '../metadata/MetadataForm';
+import DigitalObjectSummary from '../DigitalObjectSummary';
+
+function DigitalObjectNewForm() {
+  const { projectStringKey, digitalObjectType } = useParams();
+
+  // Retrieve data and set data
+  const { loading: projectLoading, error: projectError, data: projectData } = useQuery(
+    getProjectQuery, { variables: { stringKey: projectStringKey } },
+  );
+
+  if (projectLoading) return (<></>);
+  if (projectError) return (<GraphQLErrors errors={projectError} />);
+  const { project: { stringKey, displayLabel } } = projectData;
+
+  const initialDigitalObject = {
+    serializationVersion: '1',
+    digitalObjectType,
+    primaryProject: { stringKey, displayLabel },
+    otherProjects: [],
+    dynamicFieldData: {},
+    identifiers: [],
+  };
+
+  return (
+    <>
+      <ContextualNavbar
+        title={`New ${capitalize(initialDigitalObject.digitalObjectType)}`}
+        rightHandLinks={[{ link: '/digital_objects', label: 'Back to Digital Objects' }]}
+      />
+      <Can I="create_objects" of={{ subjectType: 'Project', stringKey }} passThrough>
+        {
+          (can) => {
+            if (can) {
+              return (
+                <>
+                <DigitalObjectSummary digitalObject={initialDigitalObject} />
+                <MetadataForm formType="new" digitalObject={initialDigitalObject} />
+                </>
+              );
+            }
+
+            return 'Invalid permissions for this view.';
```

Comment:
Could we make this entire route protected? or is it important to display this message in this context?

Reply:
The reason I didn't do it at a higher level (in DigitalObjects.js) is because it made it possible to avoid yet another Digital Object retrieval request at that level.

You won't actually get to this particular route unless you manually navigate to it though.  Normally the only method is by submitting the project/digital_object_type select form in the DigitalObjectNew.  But if someone did somehow type in the URL manually, I added this <Can> check to hide the object building form if they don't have the correct project rights (since a request for the project is made in this component).

I think the alternative to this <Can> would be to add a <Switch>/<ProtectedRoute> filter to this component's render content, but I think it would do effectively the same thing?

Reply:
I might be missing something, but I don't think you need to retrieve a digital object to check permissions for this route. You only need the project string key and that's passed through as a parameter.

Reply:
Ah!  You're right.   I was thinking about an earlier implementation of <DigitalObjectProtectedRoute>, but even in the latest version it only needs project string keys.  In that case, I could do that check on a higher level in DigitalObject.js.  Will do, thanks!